website/blog/2021-02-04-raptorx.md

```diff
@@ -0,0 +1,214 @@
+
+The caching mechanism is to align each read into 1MB chunks, where 1MB is configurable to be adapted to different storage capability. For example, suppose Presto issues a read with 3MB in length starting with offset 0, then Alluxio cache checks if 0 - 1MB, 1 - 2MB, and 2 - 3MB chunks are already on disk and only fetch those that are not cached. The purging policy is based on LRU. It removes chunks from a disk that has not been accessed for the longest time. The Alluxio data cache exposes a standard Hadoop File System interface to the Hive connector, transparently storing requested chunks in a high-performance, highly concurrent, and fault-tolerant storage engine which is designed to serve workloads at Facebook scale.
+
+
+
+![Alluxio Data Cache](/img/blog/2021-02-04-raptorx/alluxio.png)
+
+
+
+
+## Soft Affinity Scheduling
+
+To maximize the cache hit rate on workers, the coordinator needs to schedule the requests of the same file to the same worker. Because there is a high chance part of the file has already been cached on that particular worker. The scheduling policy is "soft", meaning that if the destination worker is too busy or unavailable, the scheduler will fallback to its secondary pick worker for caching or just skip the cache when necessary. The same [earlier post](https://prestodb.io/blog/2020/06/16/alluxio-datacaching) has a detailed explanation of the scheduling policy. The scheduling policy guarantees that cache is not on the critical path but still can boost performance.
+
+
+
+## Performance
+
+RaptorX cache has been fully deployed and battle tested within Facebook. To compare the performance with vanilla Presto, we ran TPC-H benchmark on a 114-node cluster. Each worker has a 1TB local SSD with 4 threads configured per task. We prepared TPC-H tables with a scale factor of 100 in remote storage. The following chart shows the comparison between Presto and Presto with the hierarchical cache.
+
+
+
+![Alluxio Data Cache](/img/blog/2021-02-04-raptorx/tpch.png)
+
+
+
+
+From the benchmark, scan-heavy or aggregation-heavy queries like Q1, Q6, Q12 - Q16, Q19, and Q22 all have more than 10X latency improvement. Even join-heavy queries like Q2, Q5, Q10, or Q17 have 3X - 5X latency improvements.
+
+
+
+## User Guide
+It is required to have local SSDs for workers in order to fully enable this feature. To turn on various layers of the caches in this post, tune the following configs accordingly.
+
+Scheduling (`/catalog/hive.properties`):
```

Comment:
I much appreciate the improved documentation, but it's an anti-pattern for OSS projects to have the documentation in the blog (seriously a lot of projects ended up this way). Hopefully this is just temporary?

Reply:
That is a good point. I will move it to the doc part